File: Backend/Python/Translator/application/generic/domainObjects/responses/wsBaseResponse.py


class WSBaseResponse:

    def __init__(self):
        self._status = None
        self._response = None
        self._success_message = None
        self._error_message = None

    @property
    def status(self):
        return self._status

    @status.setter
    def status(self, status):
        self._status = status

    @property
    def response(self):
        return self._response

    @response.setter
    def response(self, response):
        self._response = response

    @property
    def success_message(self):
        return self._success_message

    @success_message.setter
    def success_message(self, success_message):
        self._success_message = success_message

    @property
    def error_message(self):
        return self._error_message

    @error_message.setter
    def error_message(self, error_message):
        self._error_message = error_message

    def save_data(self, validated_data):
        self.status = validated_data.get('status')
        self.response = validated_data.get('response')
        self.success_message = validated_data.get('success_message')
        self.error_message = validated_data.get('error_message')

File: Frontend/Translator/src/integration/apiTranslate/domainObjects/responses/WSBaseResponse.js
/*
 * Licensed To: ThoughtExecution
 * Authored By: Rishi Raj Bansal
 * Developed in: December 2019
 *
*/

import domainObjectConstants from '../../constants/domainObjectConstants.js';
import FormBaseVO from '../../../../controller/formVOs/FormBaseVO';

class WSBaseResponse {

    constructor(apiResponseObject) {
        this.status = apiResponseObject[domainObjectConstants.DO_RESPONSE_BASE_STATUS];
        this.response = apiResponseObject[domainObjectConstants.DO_RESPONSE_BASE_RESPONSE];
        this.successMessage = apiResponseObject[domainObjectConstants.DO_RESPONSE_BASE_SUCCESS_MESSAGE];
        this.errorMessage = apiResponseObject[domainObjectConstants.DO_RESPONSE_BASE_ERROR_MESSAGE];
    }

    convertToFormVO(viewRequestObject){
        viewRequestObject[FormBaseVO.PARAM_BASE_FIELD_STATUS] = this.status;
        viewRequestObject[FormBaseVO.PARAM_FIELD_RESPONSE_TYPE] = this.response;
        viewRequestObject[FormBaseVO.FORM_ATTR_SUBMIT_SUCCESS_MESSAGE][FormBaseVO.PARAM_MESSAGE_FIELD_SUCCESS_MESSAGE] = this.successMessage;
        viewRequestObject[FormBaseVO.FORM_ATTR_SUBMIT_ERROR_MESSAGE][FormBaseVO.PARAM_MESSAGE_FIELD_ERROR_MESSAGE] = this.errorMessage;

    }

}


export default WSBaseResponse;
